### DISCUSSION TITLE
ks_pplns_v0.1

### DISCUSSION THREAD

**Discussion by onemorebsmith** [onemorebsmith/kaspa-stratum-bridge#68] [Announcements]
# ks_pplns v0.1

Prototype client that allows mining to a local node but receiving payouts to the parent pool [ghost-pool.io](https://grafana.ghost-pool.io/d/r_YOhXOVz/pplns-pool)

By default this will work virtually identically to the ks_bridge. It'll listen for stratum connections on port :5555 and look for a kaspad node at localhost:16110

# How it works:

On launch the client will register with the parent pool and receive a pool mining wallet address. This address is used as the coinbase wallet for all blocks mined by the client. As the client accumulates shares, they are periodically submitted to back to the parent pool, who does server-side validation and then logs the shares as normal.

Because mining is done to the local node, there is virtually no latency for mining. Latency for share submission is only used for payout calculations, which is not time-critical as they're done periodically.

<hr /><em>This discussion was created from the release <a href='https://github.com/onemorebsmith/kaspa-stratum-bridge/releases/tag/ks_pplns_v0.1'>ks_pplns_v0.1</a>.</em>

**Comment by jvxis** [comment]
Hi, @onemorebsmith great idea. How and where will my miner track their performance? I'm using the monitoring today, and all of the miners are solo mining. How can I keep the solo mining and offer the ppnls as well? Can I have an instance of the monitoring locally?

**Comment by RulezZzOr** [comment]
i am getting error during compose : 
Error response from daemon: pull access denied for onemorebsmith/kaspa_pplns, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**Reply by onemorebsmith** [reply]
try `onemorebsmith/kaspa_pplns:latest` it usually works w/o the tag but I guess not this time? I'll look at fixing that today

**Comment by jvxis** [comment]
Did anyone test it with dual mining? A miner said that it didn't work with dual. KAS + ETC or KAS + ERG.

**Reply by onemorebsmith** [reply]
Haven't tested myself personally, I'll look at trying that today
